File: backend/src/main/scala/domains/usergroup/application/output/UserGroupSummary.scala
package domains.usergroup.application.output

import domains.usergroup.model.*

import domains.user.model.Username

import java.time.Instant

final case class UserGroupSummary(
  id: UserGroupId,
  slug: UserGroupSlug,
  name: UserGroupName,
  description: UserGroupDescription,
  ownerUsername: Username,
  createdAt: Instant,
  updatedAt: Instant
)

File: frontend/src/features/usergroup/http/response/UserGroupSummary.ts
import type { Username } from '@/features/user/model/UserValues'
import type { UserGroupDescription } from '@/features/usergroup/model/UserGroupDescription'
import type { UserGroupId } from '@/features/usergroup/model/UserGroupId'
import type { UserGroupName } from '@/features/usergroup/model/UserGroupName'
import type { UserGroupSlug } from '@/features/usergroup/model/UserGroupSlug'
import type { AuditFields } from '@/shared/model/AuditFields'

export type UserGroupSummary = AuditFields & {
  id: UserGroupId
  slug: UserGroupSlug
  name: UserGroupName
  description: UserGroupDescription
  ownerUsername: Username
}
